src/platform/magento2/order/orderProcessor.js

```diff
@@ -0,0 +1,302 @@
+import {productsEquals} from 'vsf-utilities';
+import order from '../../../api/order';
+const orderUtilities = require('./order-utilities');
+const apiUtilities = require('./api-utilities');
+const config = require('config');
+const redis = require('../../../lib/redis');
+const redisClient = redis.getClient(config)
+const logger = console;
+const countryMapper = require('../../../lib/countrymapper')
+
+const TOTAL_STEPS = 4;
+var currentStep = 1;
+
+/**
+ * @param {json} orderData order data in format as described in '../models/order.md'
+ * @returns {Promise}
+ */
+async function cartIdPrepare (orderData) {
+  const api = apiUtilities.getApi(orderData.store_code);
+  const isThisAuthOrder = orderUtilities.isThisAuthOrder(orderData);
+  const {
+    cart_id: cartId,
+    user_id: userId
+  } = orderData;
+
+  return isThisAuthOrder ? api.cart.create(null, userId) : (cartId ? new Promise((resolve, reject) => {
```

Comment:
Instead of `new Promise` construct you could use shortcut - `Promise.resolve(cartId)`.

I do not like this nested ternary operators. Don't you think it could be much more readable without ternaries?
Look:
```js
if (isThisAuthOrder) {
  return api.cart.create(null, userId);
}

return cartId
  ?  Promise.resolve(cartId)
  :  api.cart.create(null);
```

Reply:
It is. I didn't take a look into this one.